pkg/resources/pki/certs.go

```diff
@@ -0,0 +1,228 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pki
+
+import (
+	"context"
+	"fmt"
+
+	banzaicloudv1alpha1 "github.com/banzaicloud/kafka-operator/api/v1alpha1"
+	"github.com/banzaicloud/kafka-operator/pkg/certutil"
+	"github.com/banzaicloud/kafka-operator/pkg/resources/kafka"
+	"github.com/banzaicloud/kafka-operator/pkg/resources/templates"
+	certv1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// A full PKI for Kafka
+
+func (r *Reconciler) kafkapki() ([]runtime.Object, error) {
+	rootCertMeta := templates.ObjectMeta(fmt.Sprintf(brokerCACertTemplate, r.KafkaCluster.Name), labelsForKafkaPKI(r.KafkaCluster.Name), r.KafkaCluster)
+	rootCertMeta.Namespace = "cert-manager"
+
+	if r.KafkaCluster.Spec.ListenersConfig.SSLSecrets.Create {
```

Comment:
If the `sslSecrets` isn't defined in CR because you want to try without ssl, this will be a nil pointer dereference.

Reply:
Good catch, I think I might have that problem in the end of `pki.go` also. Will fix.

Reply:
I fixed this by just returning at the beginning of the Reconcile if `sslSecrets` is nil. I also found an issue with when I took out a bunch of hardcoded strings and made them constants. I left one wrong by mistake.